src/components/auth/Form/AuthForm/index.tsx

```diff
@@ -75,106 +83,154 @@ const AuthForm = () => {
     },
   });
 
+  const handleVerifyCodeValidate = async (value: string) => {
+    if (value.length === 6) {
```

Comment:
```suggestion
  if (value.length === 6) {
    sendCode({ email, code: value });
    return true;
  }
```
비동기 처리와 에러 핸들링이 `useMutation`에서 처리되어서, await과 try-catch를 사용하지 않아도 될 것 같습니다.

Reply:
피드백 감사합니다!

Reply:
이 부분에 try-catch문은 아직 남아있는것 같습니다😂

Reply:
@iamkjw77 지원님! 문의드릴 부분이 있습니다.

말씀해주신 부분은 `react-hook-form`의 **validate** 속성에 매핑된 함수인데요,
제가 알기로는 **validate** 함수는 `true`나 `false(또는 문자열)`를 반환해야 하는 것으로 알고 있습니다.

에러 핸들링은 mutation 쪽에서 진행되더라도 실질적으로 true나 false(또는 문자열)를 반환할 수 없기 때문에
이 함수 안에서 try-catch 문을 사용하는 게 맞지 않을까요?

혹시 제가 잘못 이해한 부분이 있다면 알려주시면 감사하겠습니다! 😸

Reply:
넵 말씀하신대로 **validate** 함수는 `true`나 `false(또는 문자열)`를 반환해야 하는게 맞습니다!
하지만, useMutation에서 onError에서 이미 에러에 대한 처리를 하고 있어서 try-catch문은 불필요해 보입니다🤔

그래서 첫 번째 리뷰에서 말씀드린 것처럼 `return true;`를 반환하거나, 

```suggestion
    if (value.length === 6) {
      sendCode({ email, code: value });
      return true;
  }
```

아니면 말씀하신 것처럼 **true**가 아니면 문자열을 반환해도 좋을 것 같습니다☺️

```suggestion
    if (value.length === 6) {
      sendCode({ email, code: value });
      return true;
    }

    return AUTH_ERROR_MSG.CERTNO_PATTERN;
```

Reply:
![2025-01-06 16;21;49](https://github.com/user-attachments/assets/1eb66d98-a18d-4a7d-a852-c910f422c4cd)
> **mutation 함수** 자체가 비동기 함수이기 때문에 **async-await 키워드를 제외**시키는게 맞고
> 또한, `react-query`가 **onError**에서 에러 처리를 하기 때문에 **try-catch를 제외**하는게 맞다.

여기까지는 이해가 됐습니다! 

지원님께서 제안하신 코드에 의문이 드는 게 mutation 함수 자체가 비동기 함수이기 때문에 validate에 매핑된 함수에서 `return true`를 하게 되면 **mutation 함수의 성공 여부와는 상관 없이 무조건 true를 반환**하게 되고, 이후 `onError`에서 에러 메시지를 띄우는 구조가 됩니다. (첨부 이미지 참고)

이런 방식이 올바른 접근인지, 혹은 더 나은 방식이 있는지 궁금합니다!

Reply:
저는 단순히 useMutation의 결과만 가지고 처리하는 경우를 생각해서, onSuccess와 onError로 충분히 처리 가능할 것이라고 생각했습니다. 그래서 추가적인 try-catch가 불필요할 것이라고 생각했구요! 

하지만, 말씀하신 것처럼 mutation 함수의 성공 여부와는 상관 없이 무조건 true를 반환해서, onError에서 에러를 띄우기 전까지 텀이 생기는 문제가 있네요🤔

이 경우는 처음에 작성하신 것처럼 **async, await**을 사용하는 것이 좋을 것 같습니다! 
**sendCode** 자체는 Promise를 반환하지 않아서 아래처럼 Promise 객체를 만들어서 작성해 보았습니다. 

```suggestion
   const handleVerifyCodeValidate = async (value: string) => {
     if (value.length !== 6) return false;

    return new Promise<boolean | string>((resolve) => {
      sendCode(
        { email, code: value },
        {
          onSuccess: () => resolve(true),
          onError: () => resolve(AUTH_ERROR_MSG.CERTNO_PATTERN),
        },
      );
    });
  };
```

Reply:
제안해주신 코드 실행해보니 깔끔하게 작동합니다.

해당 필드 `rules`의  `minLength`, `maxLength` 가 만족하면
`validate` 함수가 실행되는 거 같아서 분기문은 제외했습니다!

감사합니닷❗❗ 😃